sql/idempotent/005-vectorizer.sql

```diff
@@ -0,0 +1,1178 @@
+
+
+-------------------------------------------------------------------------------
+-- execute_async_ext_vectorizer
+create or replace function ai.execute_async_ext_vectorizer(vectorizer_id int) returns void
+as $python$
+    #ADD-PYTHON-LIB-DIR
+    import ai.vectorizer
+    ai.vectorizer.execute_async_ext_vectorizer(plpy, vectorizer_id)
+$python$
+language plpython3u volatile security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- _vectorizer_ext_delete
+create or replace function ai._vectorizer_ext_delete(vectorizer_id int) returns void
+as $python$
+    #ADD-PYTHON-LIB-DIR
+    import ai.vectorizer
+    ai.vectorizer.delete_vectorizer(plpy, vectorizer_id)
+$python$
+language plpython3u volatile security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- indexing_none
+create or replace function ai.indexing_none() returns jsonb
+as $func$
+    select jsonb_build_object('implementation', 'none')
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- indexing_diskann
+create or replace function ai.indexing_diskann
+( min_rows int default 100000
+, storage_layout text default null
+, num_neighbors int default null
+, search_list_size int default null
+, max_alpha float8 default null
+, num_dimensions int default null
+, num_bits_per_dimension int default null
+) returns jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'diskann'
```

Comment:
I like this idea.

Reply:
cfd65d6d02a85cf44acec6e609ea826508731249

Reply:
I was thinking about something similar, but instead of using a new field, can we just create a new type, like chunking_config, embedding_config.

This is what the `by_range` function from timescale does, it returns a `dimension_info`